File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        for(int i=0; i<clothes.length; i++){
            String category = clothes[i][1];
            if(map.get(category) == null){
                map.put(category, 1);
            }else{
                map.put(category, map.get(category)+1);
            }
        }
        
        int answer = 1;
        
        for(Map.Entry<String, Integer> el : map.entrySet()){
            
            answer *= (el.getValue()+1);
        }               
        
        answer--;
        
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
from itertools import combinations

def solution(clothes):
    answer = 1
    
    closet = {}
    
    for c in clothes:
        if closet.get(c[1]) == None:
            closet[c[1]] = 1
        else:
            closet[c[1]] +=1
    
    for i in closet.values():
        answer *= (i+1)
    
    answer -= 1
       
    return answer